### PR TITLE
Add support for Emqx Container

### PR DESCRIPTION
**Pull request by andreaswillems**

### CODE REVIEW

**Comment by jarlah** [comment]
will take a look at it :) Thanks for the contribution

File: mix.exs [hunk 1]
```diff
@@ -2,7 +2,7 @@ defmodule TestcontainersElixir.MixProject do
   use Mix.Project
 
   @app :testcontainers
-  @version "1.7.0"
+  @version "1.7.1"
```
Comment (jarlah) [inline mix.exs:5]: there is a workflow step that verifies if the version is consistent. you need to update https://github.com/testcontainers/testcontainers-elixir/blob/main/lib/util/constants.ex#L5 as well

File: lib/container/emqx_container.ex [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+          EmqxContainer.host(),
+          config.dashboard_port,
+          config.wait_timeout,
+          1000
+        )
+      ]
+
+    @impl true
+    # TODO Implement the `is_starting/3` function for the `ContainerBuilder` protocol.
+    def is_starting(_config, _container, _conn), do: :ok
```
Comment (jarlah) [inline lib/container/emqx_container.ex:148]: i think the wording of this function is a bit wrong. Its actually an "on startup hook" function. Its called when the container "is starting". Used for executing scripts right after container has started. see https://github.com/testcontainers/testcontainers-elixir/blob/main/lib/container/kafka_container.ex#L177

Comment (jarlah) [reply lib/container/emqx_container.ex:148]: Just a FYI, i should think about the naming of the function. If you have a better name feel free to suggest

Comment (jarlah) [reply lib/container/emqx_container.ex:148]: However, i see now the docs in https://github.com/testcontainers/testcontainers-elixir/blob/main/lib/container/kafka_container.ex#L177 is wrong. Its actually not a function that "checks if container is starting" as the kafka implementation says it is.

Comment (jarlah) [reply lib/container/emqx_container.ex:148]: added new issue in https://github.com/testcontainers/testcontainers-elixir/issues/84

**Review by jarlah — CHANGES_REQUESTED** [review]
Seems build steps are working. Fix the version in constants and i can take a final look after that

File: lib/container/emqx_container.ex (hunk 2 above)
Comment (andreaswillems) [reply lib/container/emqx_container.ex:148]: @jarlah Thanks for your review.

Looking at the startup procedure of the containers, it seems that having the function `is_starting` is redundant, see https://github.com/testcontainers/testcontainers-elixir/blob/b0edf48e5d1a31f13eb312904f5f9c6a94536106/lib/testcontainers.ex#L218.

There are two real implementations, in the aforementioned Kafka container and in the supporting Nginx container (https://github.com/testcontainers/testcontainers-elixir/blob/b0edf48e5d1a31f13eb312904f5f9c6a94536106/test/support/nginx_container.ex#L17).

My suggestion would be to remove this function from the protocol and move the existing implementations into a after-startup hook.

Comment (jarlah) [reply lib/container/emqx_container.ex:148]: we can consider it. Leaving it out of this PR naturally.

File: test/container/emqx_container_test.exs [hunk 3]
```diff
@@ -0,0 +1,48 @@
+defmodule Testcontainers.Container.EmqxContainerTest do
```
Comment (jarlah) [inline test/container/emqx_container_test.exs:1]: you aren't actually doing stuff with the container. Do you think we should add a test to verify we can actually use the emqx container for something "useful" ?

Comment (andreaswillems) [reply test/container/emqx_container_test.exs:1]: Good idea. I will try and come up with something.

Comment (jarlah) [reply test/container/emqx_container_test.exs:1]: rebase and fix conflicts. and i can merge it with and you can add test later in a second PR, if you agree

**Review by jarlah — APPROVED** [review]

**Pull request merged**